Adding "Include UOM" in Reports with Qty and Rates

SaiFi0102 opened the pull request:
**Problem**: Stock reports only show in stock UOM and does not show other UOMs
**Solution**: Include a filter field "Include UOM" and add extra columns for converted rate and qty columns using conversion_factor

**Thoughts for the future**: Maybe use MultiSelect for Include UOM field.
**Missing out on**: Due to the complexity of getting conversion_factors from UOM Conversion Factor DocType, I did not consider those conversion factors. Should the reports also convert using UOM Conversion Factor?

**Implemented on reports**: Stock Ledger, Stock Balance, Stock Projected Qty 
**Further implementation**: Since qty and rate fields are included in many reports, I will continue updating reports to include the "Include UOM" field whenever the need arises. I would encourage the community to update the rest of the reports.

SaiFi0102 committed:
Added Include UOM field for Stock Balane, Stock Ledger and Stock Projected Qty

SaiFi0102 committed:
Add columns in result list-of-lists instead of converting reports to list-of-dicts

rmehta commented on erpnext/stock/report/stock_balance/stock_balance.py:
```diff
@@ -72,17 +79,17 @@ def get_columns():
 		_("Warehouse")+":Link/Warehouse:100",
 		_("Stock UOM")+":Link/UOM:90",
-		_("Opening Qty")+":Float:100",
+		{"label": _("Opening Qty"), "fieldname": "opening_qty", "fieldtype": "Float", "width": 100, "convertible": "qty"},
```
+1 for using the new format, but why not refactor all the columns, so that it looks consistent

rmehta commented on erpnext/stock/report/stock_ledger/stock_ledger.py:
```diff
@@ -103,6 +111,14 @@ def get_item_details(items, sl_entries):
 			item_details.setdefault(item.name, item)
 
+	if include_uom:
+		for item in frappe.db.sql("""select item.name, ucd.conversion_factor from `tabItem` item
```
is this the same as above? (commonify)

rmehta commented on erpnext/stock/utils.py:
```diff
@@ -248,3 +248,35 @@ def validate_warehouse_company(warehouse, company):
+		return
+
+	convertible_cols = {}
+	for iCol in reversed(range(0, len(columns))):
```
snake_case in Python! `col_idx`

SaiFi0102 committed:
For requested changes
-Merged conversion factor query with item detail queries
-Ensuring snail_case
-Made columns consistently list-of-dicts

SaiFi0102 commented:
@rmehta can you open the PR again? I've already completed the required changes.
